### PR TITLE
Replace use of scipy.fftpack with scipy.fft

### PR DESCRIPTION
**Pull request by mdhaber**
`scipy.fftpack` is considered legacy; this updates uses of `scipy.fftpack` to use `scipy.fft`, instead. To avoid unnecessary churn, I didn't modify filenames that include `fftpack`.

### CODE REVIEW

File: intro/scipy/examples/plot_fftpack.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
 period = 5.0
 
 time_vec = np.arange(0, 20, time_step)
-sig = np.sin(2 * np.pi / period * time_vec) + 0.5 * np.random.randn(time_vec.size)
+sig = np.sin(2 * np.pi / period * time_vec) + 0.5 * rng.randn(time_vec.size)
```
Comment (mdhaber) [inline intro/scipy/examples/plot_fftpack.py:31]: There are many instances where an Generator is available but is unused. I'll create a separate PR for that.

File: intro/scipy/examples/solutions/plot_image_blur.py [hunk 2]
```diff
@@ -41,14 +41,14 @@
 #####################################################################
 
 # Padded fourier transform, with the same shape as the image
-# We use :func:`scipy.signal.fftpack.fft2` to have a 2D FFT
-kernel_ft = sp.fftpack.fft2(kernel, shape=img.shape[:2], axes=(0, 1))
+# We use :func:`scipy.signal.fft.fft2` to have a 2D FFT
```
Comment (rossbar) [inline intro/scipy/examples/solutions/plot_image_blur.py:44]:
```suggestion
# We use :func:`scipy.fft.fft2` to have a 2D FFT
```

**Review by rossbar — APPROVED** [review]
Thanks @mdhaber! +1 for updating the references away from fftpack + the other fixups here

File: intro/scipy/examples/solutions/plot_image_blur.py (hunk 2 above)
Comment (mdhaber) [reply intro/scipy/examples/solutions/plot_image_blur.py:44]: Ah didn't notice. Will commit before this is merged.

File: intro/scipy/examples/plot_fftpack.py (hunk 1 above)
Comment (mdhaber) [reply intro/scipy/examples/plot_fftpack.py:31]:
```suggestion
sig = np.sin(2 * np.pi / period * time_vec) + 0.5 * rng.normal(size=time_vec.size)
```

**Review by jarrodmillman — APPROVED** [review]

**Pull request merged**